cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -868,87 +811,95 @@ impl MachInst for Inst {
         }
     }
 
-    fn gen_jump(blockindex: BlockIndex) -> Inst {
-        Inst::jmp_known(BranchTarget::Block(blockindex))
+    fn gen_jump(label: MachLabel) -> Inst {
+        Inst::jmp_known(BranchTarget::Label(label))
+    }
+
+    fn gen_constant(to_reg: Writable<Reg>, value: u64, _: Type) -> SmallVec<[Self; 4]> {
+        let mut ret = SmallVec::new();
+        let is64 = value > 0xffff_ffff;
```

Comment:
Is this definition of `is64` right?  That seems like it's an unsigned criterion, but the general rule on Intel for 32-bit immediate fields is that they are sign extended to 64 bits as appropriate.  If this logic simply moved from elsewhere in this patch, then leave it as is; but otherwise maybe change to the signed variant, using `low32willSXto64` ?

Reply:
I think so, or at least, this was existing logic in the `Iconst` lowering prior to this patch:

https://github.com/bytecodealliance/wasmtime/blob/a75377565f830052094aa8aa72c5e7a6b787fa18/cranelift/codegen/src/isa/x64/lower.rs#L116